include/boost/mp11/algorithm.hpp

```diff
@@ -95,6 +95,15 @@ struct list_size_mismatch
 {
 };
 
+#if BOOST_WORKAROUND( BOOST_CUDA_VERSION, >= 9000000 ) && BOOST_WORKAROUND( BOOST_CUDA_VERSION, < 10000000 )
```

Comment:
We usually use

BOOST_WORKAROUND( BOOST_CUDA_VERSION, >= 9000000 && BOOST_CUDA_VERSION < 10000000 )

for those (the two macro arguments are concatenated so you could use an expression.)

There's also the cuter

BOOST_WORKAROUND( BOOST_CUDA_VERSION, / 1000000 == 9 )

although this doesn't update well to a range of versions (in case 10 also needs the fix.)

Reply:
Oh, okay, I was not sure how `BOOST_WORKAROUND` was exactly working in the background. :) I will go for the second version as we need to adjust this anyway if nvidia doesn't fix this bug for version 10. :wink: